components/catalogs/back/src/Infrastructure/Persistence/Catalog/Product/IsValidCatalogProductQuery.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Infrastructure\Persistence\Catalog\Product;
+
+use Akeneo\Catalogs\Application\Persistence\Catalog\GetCatalogProductSelectionCriteriaQueryInterface;
+use Akeneo\Catalogs\Application\Persistence\Catalog\Product\IsValidCatalogProductQueryInterface;
+use Akeneo\Pim\Enrichment\Component\Product\Query\Filter\Operators;
+use Akeneo\Pim\Enrichment\Component\Product\Query\ProductQueryBuilderFactoryInterface;
+use Doctrine\DBAL\Connection;
+use Ramsey\Uuid\Uuid;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class IsValidCatalogProductQuery implements IsValidCatalogProductQueryInterface
```

Comment:
```suggestion
class IsProductInCatalogQuery implements IsValidCatalogProductQueryInterface
```
I didn't like the name because I didn't understand what it does at first sight. What do you think about these names?
- IsProductInCatalogQuery
- IsProductBelongingToCatalogQuery
- IsCatalogContainingProductQuery